packages/suite-desktop-core/e2e/support/pageActions/settingsActions.ts

```diff
@@ -1,85 +1,98 @@
+    readonly coinAdvanceSettingsButton = (coin: NetworkSymbol) =>
+        this.window.getByTestId(`@settings/wallet/network/${coin}/advance`);
+    readonly coinBackendSelector: Locator;
+    readonly coinBackendSelectorOption = (backend: BackendType) =>
+        this.window.getByTestId(`@settings/advance/${backend}`);
+    readonly coinAddressInput: Locator;
+    readonly coinAdvanceSettingSaveButton: Locator;
+
     constructor(window: Page) {
         this.window = window;
+        this.settingsHeader = this.window.getByTestId('@settings/menu/title');
+        this.debugTabButton = this.window.getByTestId('@settings/menu/debug');
+        this.applicationTabButton = this.window.getByTestId('@settings/menu/general');
+        this.deviceTabButton = this.window.getByTestId('@settings/menu/device');
+        this.coinsTabButton = this.window.getByTestId('@settings/menu/wallet');
+        this.earlyAccessJoinButton = this.window.getByTestId('@settings/early-access-join-button');
+        this.earlyAccessConfirmCheck = this.window.getByTestId(
+            '@settings/early-access-confirm-check',
+        );
+        this.earlyAccessConfirmButton = this.window.getByTestId(
+            '@settings/early-access-confirm-button',
+        );
+        this.earlyAccessSkipButton = this.window.getByTestId('@settings/early-access-skip-button');
+        this.settingsCloseButton = this.window.getByTestId('@settings/menu/close');
+        this.modal = this.window.getByTestId('@modal');
+        //coin Advance settings
+        this.coinBackendSelector = this.window.getByTestId('@settings/advance/select-type/input');
+        this.coinAddressInput = this.window.getByTestId('@settings/advance/url');
+        this.coinAdvanceSettingSaveButton = this.window.getByTestId(
+            '@settings/advance/button/save',
+        );
     }
 
     async toggleDebugModeInSettings() {
-        const settingsHeader = this.window.getByTestId('@settings/menu/title');
-        await settingsHeader.waitFor({ state: 'visible', timeout: 10_000 });
-        const timesClickToSetDebugMode = 5;
-        for (let i = 0; i < timesClickToSetDebugMode; i++) {
-            await settingsHeader.click();
+        await expect(this.settingsHeader).toBeVisible();
+        for (let i = 0; i < this.TIMES_CLICK_TO_SET_DEBUG_MODE; i++) {
+            await this.settingsHeader.click();
         }
-        await this.window.getByTestId('@settings/menu/debug').waitFor({ state: 'visible' });
+        await expect(this.debugTabButton).toBeVisible();
     }
 
-    async goToSettingSection(section: Section) {
-        await this.window.getByTestId(`@settings/menu/${section}`).click();
-        //TODO: #15552 Refactor navigation verification to specific tests and remove this method completely
-        await this.window
-            .getByTestId(settingSectionsLocators[section])
-            .waitFor({ state: 'visible', timeout: 10_000 });
-    }
```

Comment:
I have spent like 30 minute thinking whether keep it or not. But in the end the implemenetation of this method grow way too much due to locator rafactor. 
Anyway every test that does the navigation, then interact with element on the page. So the verification was duplicate and was checking just one random element on that section. So I completely replaced it with just click on tab on test level